### PR TITLE
fix: Language Server analytics permitted function implemented

### PR DESCRIPTION
**Pull request by acke**
### Description

Since we are having more geographically distributed environments, we should only send analytics to the MT US environments, excluding FedRAMP.

### Checklist

- [x] Tests added and all succeed
- [x] Linted

🚨After having merged, please update the CLI go.mod to pull in latest language server.

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

https://snyksec.atlassian.net/browse/IDE-24

### CODE REVIEW

File: application/config/config.go [hunk 1]
```diff
@@ -872,3 +875,16 @@ func (c *Config) IdeName() string {
 func (c *Config) IsFedramp() bool {
 	return c.Engine().GetConfiguration().GetBool(configuration.IS_FEDRAMP)
 }
+
+func (c *Config) IsAnalyticsPermitted() bool {
+	u, err := url.Parse(c.Engine().GetConfiguration().GetString(configuration.API_URL))
+
+	if err != nil {
+		log.Trace().Err(err).Msg("unable to parse configured API_URL")
```
Comment (bastiandoetsch) [inline application/config/config.go:883]: Please use `c.Logger().With()...`

File: application/config/config.go [hunk 2]
```diff
@@ -872,3 +875,16 @@ func (c *Config) IdeName() string {
 func (c *Config) IsFedramp() bool {
 	return c.Engine().GetConfiguration().GetBool(configuration.IS_FEDRAMP)
 }
+
+func (c *Config) IsAnalyticsPermitted() bool {
+	u, err := url.Parse(c.Engine().GetConfiguration().GetString(configuration.API_URL))
```
Comment (bastiandoetsch) [inline application/config/config.go:880]: Initialization order may be interesting here - is the endpoint already set when IsAnalyticsPermitted is called?

Comment (acke) [reply application/config/config.go:882]: We went through the server's `initlialize` handler, and made sure that all calls to `IsAnalyticsPermitted` happen after the API URL has been set in the configuration 👍  `updateTelemetry` is the first function inside the `initialize` handler that is checking if analytics are permitted, and it's being called after setting the API URL. Further calls are done in the `initialized` handler, which also happens afterwards.

File: application/config/config.go [hunk 3]
```diff
@@ -67,12 +67,16 @@ const (
...
-	currentConfig      *Config
-	mutex              = &sync.Mutex{}
-	LicenseInformation = "License information\n FILLED DURING BUILD"
+	Version                        = "SNAPSHOT"
+	LsProtocolVersion              = "development"
+	Development                    = "true"
+	currentConfig                  *Config
+	mutex                          = &sync.Mutex{}
+	LicenseInformation             = "License information\n FILLED DURING BUILD"
+	analyticsPermittedEnvironments = map[string]bool{
```
Comment (bastiandoetsch) [inline application/config/config.go:76]: Could be beneficial to store these in the Go Application Framework config (gafConfig). As this applies also to CLI.

**Review by bastiandoetsch — APPROVED** [review]
Nice work. Marked an optional todo - but can be merged.

**Pull request merged**